src/main/java/com/powsybl/network/conversion/server/NetworkConversionService.java

```diff
@@ -494,21 +509,83 @@ public void setReportServerRest(RestTemplate reportServerRest) {
     public void reindexAllEquipments(UUID networkUuid) {
         Network network = getNetwork(networkUuid);
 
-        // delete all network equipments infos
-        deleteAllEquipmentInfosOnInitialVariant(networkUuid);
+        // delete all network equipments infos. deleting a lot of documents in ElasticSearch is slow, we delete the index instead in maintenance script before reindexing
+        deleteAllEquipmentInfosByNetworkUuid(networkUuid);
 
         // recreate all equipments infos
-        insertEquipmentIndexes(network, networkUuid, VariantManagerConstants.INITIAL_VARIANT_ID);
+        AtomicReference<Long> startTime = new AtomicReference<>(System.nanoTime());
+        try {
+            // save initial variant infos
+            Map<String, EquipmentInfos> initialVariantEquipmentInfos = getEquipmentInfos(network, networkUuid, VariantManagerConstants.INITIAL_VARIANT_ID);
+            equipmentInfosService.addAll(new ArrayList<>(initialVariantEquipmentInfos.values()));
+
+            // get variant ids without the initial that is already processed and is the reference
+            List<String> variantIds = network.getVariantManager()
+                    .getVariantIds()
+                    .stream()
+                    .filter(variantId -> !variantId.equals(VariantManagerConstants.INITIAL_VARIANT_ID))
+                    .toList();
+
+            for (String variantId : variantIds) {
+                // switch to working variant and change initial variant infos variantId for comparisons
+                network.getVariantManager().setWorkingVariant(variantId);
+                initialVariantEquipmentInfos.values().forEach(equipmentInfos ->
+                        equipmentInfos.setVariantId(variantId));
```

Comment:
I understand why it's done, but it could be buggy if done, later, in a @Transactional